src/test/regress/sql/gp_query_id.sql

```diff
@@ -0,0 +1,139 @@
+--
+-- Test the query command identification
+--
+-- start_ignore
+create extension if not exists gp_inject_fault;
+drop function if exists sirv_function();
+drop function if exists not_inlineable_sql_func(i int);
+drop table if exists test_data1;
+drop table if exists test_data2;
+drop table if exists t;
+drop table if exists t1;
+drop table if exists t2;
+-- end_ignore
+
+set client_min_messages = notice;
+select gp_inject_fault('all', 'reset', dbid) from gp_segment_configuration;
+
+create or replace function sirv_function() returns text as $$
+declare
+    result bool;
+begin
+    create table test_data (x int, y int) with (appendonly=true) distributed by (x);
+    insert into test_data values (1, 1);
+    select count(1) > 0 into result from test_data;
+    drop table test_data;
+    return case when result then 'PASS' else 'FAIL' end;
+end $$ language plpgsql;
```

Comment:
The function can be simplified: it can return int
```suggestion
create or replace function sirv_function() returns int as $$
declare
    result int;
begin
    create table test_data (x int, y int) with (appendonly=true) distributed by (x);
    insert into test_data values (1, 1);
    select count(1) into result from test_data;
    drop table test_data;
    return result;
end $$ language plpgsql;
```